### PR TITLE
 activate-polyfills guide

### PR DESCRIPTION
**Pull request by blackgirlbytes**
- Moved activate polyfills out of connect guide
- Updated community DWN info to match @angiejones's updates in main
- Created guide called Accessing Data with DRLs 
- Created tests for code snippets for Accessing Data with DRLs

### CODE REVIEW

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+sidebar_position: 10
+---
+
+import CodeSnippet from '@site/src/components/CodeSnippet';
+
+# Upgrade to DWA
+
+A **Decentralized Web App (DWA)** is a Progressive Web App (PWA) that incorporates Web5 functionalities, providing offline capabilities, Decentralized Resource Locators (DRLs), and offline synchronization of Decentralized Web Nodes (DWNs). For a practical implementation, refer to the [DWA starter app](https://github.com/TBD54566975/dwa-starter).
```
Comment (chris-tbd) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:9]: Link to resource to read more?

Comment (chris-tbd) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:9]: Link to DWN guide

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Import packages
+
+```js
+import { activatePolyfills } from '@web5/api';
+```
+
+## Set up a service worker
+Configure a service worker to activate Web5 polyfills while managing caching and data synchronization tasks.
```
Comment (chris-tbd) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:18]: Not totally sure what this means - is this something that devs will more inherently understand?

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+sidebar_position: 10
+---
+
+import CodeSnippet from '@site/src/components/CodeSnippet';
+
+# Upgrade to DWA
+
+A **Decentralized Web App (DWA)** is a Progressive Web App (PWA) that incorporates Web5 functionalities, providing offline capabilities, Decentralized Resource Locators (DRLs), and offline synchronization of Decentralized Web Nodes (DWNs). For a practical implementation, refer to the [DWA starter app](https://github.com/TBD54566975/dwa-starter).
+
+## Import packages
```
Comment (chris-tbd) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:11]: Before jumping into the code can we add a sentence or two of what the point of this page is?

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    return {
+      ttl: 30000,
+    };
+  },
+});
+```
+
+## Fetch data via DRLs
+
+After you [write records](/docs/web5/build/decentralized-web-nodes/write-to-dwn) to a DWN, you can Decentralized Resource Locators (DRLs) to fetch data directly from the decentralized network.
```
Comment (chris-tbd) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:33]: How are `DRLs` resolved/used? Can I just use one via my browser, or do I need a special app/library?

File: site/docs/web5/build/decentralized-web-nodes/web5-connect.mdx [hunk 5]
```diff
@@ -224,25 +224,15 @@ The outputs of this method invocation will be used throughout the other API meth
...
 Below are examples of how to use `Web5.connect()` with or without additional option parameters.
 
 ##### Connect with no parameters
 <CodeSnippet functionName="createDidAutomatically" />
 
-##### Connect to DIF/Google Community Node
+##### Connect to a Community Node Instance
 
-Connect to the community node instance hosted by Google and DIF:
+This instance hosted by Google and DIF provide a reliable staging environment for apps. Note: data on your DWN stays encrypted and in your control.
```
Comment (chris-tbd) [inline site/docs/web5/build/decentralized-web-nodes/web5-connect.mdx:235]: Do we need to explain what a Community Node is, how access/permissions work, or link to any other documentation/references?

**Review by chris-tbd — CHANGES_REQUESTED** [review]

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx (hunk 2 above)
Comment (leordev) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:18]: service workers are something that PWA developers understand: https://web.dev/learn/pwa/service-workers

Comment (blackgirlbytes) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:18]: I'll add a link to that that then.

Comment (blackgirlbytes) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx]: Thanks @chris-tbd and @leordev! I updated the doc. I tend to be a little wordy/verbose in documentation, so I was trying to be terse and get to the point, but I think I was too terse 😆 . Hopefully, this strikes a better balance.

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx (hunk 3 above)
Comment (blackgirlbytes) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:15]: done

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx (hunk 1 above)
Comment (blackgirlbytes) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx]: Linked!

Comment (blackgirlbytes) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx]: I linked to a blog post you wrote. Not too sure what if I have resources on DWAs.

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx (hunk 4 above)
Comment (blackgirlbytes) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx]: Hopefully, my updated explanation is more clear!

File: site/docs/web5/build/decentralized-web-nodes/web5-connect.mdx (hunk 5 above)
Comment (blackgirlbytes) [reply site/docs/web5/build/decentralized-web-nodes/web5-connect.mdx:235]: at first a link wasn't live but now it is, so i can add more details! thanks!

File: site/docs/web5/build/decentralized-web-nodes/web5-connect.mdx [hunk 6]
```diff
@@ -224,25 +224,16 @@ The outputs of this method invocation will be used throughout the other API meth
...
 
 ##### Connect with no parameters
 <CodeSnippet functionName="createDidAutomatically" />
 
-##### Connect to DIF/Google Community Node
+##### Connect to a Community Node Instance
+
 
-Connect to the community node instance hosted by Google and DIF:
+Use up to 1GB of free storage on a Community DWN instance hosted by [Google and the Decentralized Identity Foundation (DIF)](https://identity.foundation/dwn/) for staging environments.
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/web5-connect.mdx:236]:
```suggestion
Use up to 1GB of free storage on a Community DWN instance hosted by [Google and the Decentralized Identity Foundation (DIF)](https://identity.foundation/dwn/) for non-production environments.
```

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+## What is a DWA?
+
+A [Decentralized Web App (DWA)](/blog/dwas-vs-pwas) is a Progressive Web App (PWA) that incorporates Web5 functionalities, providing offline capabilities and offline synchronization of [Decentralized Web Nodes (DWNs)](/docs/web5/learn/decentralized-web-nodes/). 
+
+This guide will walk you through upgrading your PWA to a DWA. For a practical implementation, refer to the [DWA starter app](https://github.com/TBD54566975/dwa-starter).
+
+## Import packages
+
+Import the following package to handle interactions with Decentralized Web Nodes (DWNs) and manage offline data:
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:17]:
```suggestion
Import the following package to handle interactions with DWNs and manage offline data:
```

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    return {
+      ttl: 30000,
+    };
+  },
+});
+```
+
+## Fetch data via DRLs
+
+Decentralized Resource Locators (DRLs) enable direct access to data stored in Decentralized Web Nodes (DWNs). After writing records to a DWN, you can use DRLs (prefixed with `dweb://`) to fetch data.
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:41]:
```suggestion
Decentralized Resource Locators (DRLs) enable direct access to data stored in DWNs. After writing records to a DWN, you can use DRLs (prefixed with `dweb://`) to fetch data. 
```

Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:41]: the prefix note doesn't match the example:

```
https://dweb/${did}/read/records/${recordId
```

Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:41]:
```suggestion
Decentralized Resource Locators (DRLs) enable direct read access to permussioned data stored in DWNs. To read records from a DWN, you can use DRLs (prefixed with `dweb://`) to fetch the data. 
```

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+## Fetch data via DRLs
+
+Decentralized Resource Locators (DRLs) enable direct access to data stored in Decentralized Web Nodes (DWNs). After writing records to a DWN, you can use DRLs (prefixed with `dweb://`) to fetch data. 
+
+Here's how to fetch a record from a DWN using a DRL:
+
+<CodeSnippet snippetName="drlFetchReadRecord"/>
+
+DRLs return JSON data that can be used throughout your web app, from populating UI elements to setting attributes on HTML tags.
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:47]: should show an example below of using it inline as the `src` of an `` or something like that

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx (hunk 8 above)
Comment (blackgirlbytes) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:41]: Oh true. It should be dweb/ . Thats a typo on my part

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx [hunk 10]
```diff
@@ -0,0 +1,47 @@
+---
+sidebar_position: 10
+---
+
+import CodeSnippet from '@site/src/components/CodeSnippet';
+
+# Upgrade to DWA
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:7]:
```suggestion
# Accessing Data with DRLs
```

what do you think of something like this for the title? I'm thinking i may not be upgrading an existing app, so the current title might not grab my attention

Comment (angiejones) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:7]: if you change the title, I'd just delete the next header and go right into the text of that section

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+## What is a DWA?
+
+A [Decentralized Web App (DWA)](/blog/dwas-vs-pwas) is a Progressive Web App (PWA) that incorporates Web5 functionalities, providing offline capabilities and offline synchronization of [Decentralized Web Nodes (DWNs)](/docs/web5/learn/decentralized-web-nodes/). 
+
+This guide will walk you through upgrading your PWA to a DWA. For a practical implementation, refer to the [DWA starter app](https://github.com/TBD54566975/dwa-starter).
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:13]:
```suggestion
This guide will walk you through how to read from DWNs within your DWA. For a practical implementation, refer to the [DWA starter app](https://github.com/TBD54566975/dwa-starter).
```

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx (hunk 10 above)
Comment (angiejones) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:7]: also if you change the title, change the file name too

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx (hunk 8 above)
Comment (blackgirlbytes) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx]: done

File: site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx (hunk 10 above)
Comment (blackgirlbytes) [reply site/docs/web5/build/decentralized-web-nodes/upgrade-to-dwa.mdx:7]: I like your suggestion!

File: site/docs/web5/build/decentralized-web-nodes/access-data-with-drls.mdx [hunk 12]
```diff
@@ -0,0 +1,52 @@
+---
+sidebar_position: 10
+---
+
+import CodeSnippet from '@site/src/components/CodeSnippet';
+
+# Accessing Data with DRLs
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/access-data-with-drls.mdx:7]:
```suggestion
# Access Data with DRLs
```

File: site/docs/web5/build/decentralized-web-nodes/access-data-with-drls.mdx [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+
+A [Decentralized Web App (DWA)](/blog/dwas-vs-pwas) is a Progressive Web App (PWA) that incorporates Web5 functionalities, providing offline capabilities and offline synchronization of [Decentralized Web Nodes (DWNs)](/docs/web5/learn/decentralized-web-nodes/). 
+
+This guide will walk you through how to read from DWNs within your DWA via **Decentralized Resource Locators (DRLs)**. For a practical implementation, refer to the [DWA starter app](https://github.com/TBD54566975/dwa-starter).
+
+## Import packages
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/access-data-with-drls.mdx:13]:
```suggestion
## Import package
```

File: site/docs/web5/build/decentralized-web-nodes/access-data-with-drls.mdx [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+    return {
+      ttl: 30000,
+    };
+  },
+});
+```
+
+## Fetch data via DRLs
+
+DRLs enable direct read access to permussioned data stored in DWNs. To read records from a DWN,  After writing records to a DWN, you can use DRLs (prefixed with `https://dweb`) to fetch the data.
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/access-data-with-drls.mdx:39]:
```suggestion
Decentralized Resource Locators (**DRLs**) enable direct read access to permissioned data stored in DWNs. To read records from a DWN,  After writing records to a DWN, you can use DRLs (prefixed with `https://dweb`) to fetch the data. 
```

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/access-data-with-drls.test.js [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+  test('drl fetches a read record', async () => {
+    const mockEvent = {
+      currentTarget: {
+        files: [new Blob(['fake image data'], { type: 'image/png' })],
+      },
+    };
+    const record = await uploadImage(mockEvent)
+    recordId = record.id;
+    // :snippet-start: drlFetchReadRecord
+    const dwebUrl = `https://dweb/${did}/read/records/${recordId}`;
```
Comment (angiejones) [inline site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/access-data-with-drls.test.js:54]: can you rename to `drl`. it probably won't be clear that `dwebUrl` and `drl` mean the same thing

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/access-data-with-drls.test.js (hunk 15 above)
Comment (blackgirlbytes) [reply site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/access-data-with-drls.test.js:54]: True!

**Review by angiejones — APPROVED** [review]

**Pull request merged**